src/Simplex/Chat.hs

```diff
@@ -3364,173 +3364,135 @@ agentSubscriber = do
       where
         run action = action `catchChatError'` (toView' . CRChatError Nothing)
 
-type AgentBatchSubscribe = AgentClient -> [ConnId] -> ExceptT AgentErrorType IO (Map ConnId (Either AgentErrorType ()))
-
-subscribeUserConnections :: VersionRangeChat -> Bool -> AgentBatchSubscribe -> User -> CM ()
-subscribeUserConnections vr onlyNeeded agentBatchSubscribe user = do
+subscribeUserConnections :: VersionRangeChat -> Bool -> User -> CM ()
+subscribeUserConnections vr onlyNeeded user = do
   -- get user connections
-  ce <- asks $ subscriptionEvents . config
-  (conns, cts, ucs, gs, ms, sfts, rfts, pcs) <-
-    if onlyNeeded
-      then do
-        (conns, entities) <- withStore' (`getConnectionsToSubscribe` vr)
-        let (cts, ucs, ms, sfts, rfts, pcs) = foldl' addEntity (M.empty, M.empty, M.empty, M.empty, M.empty, M.empty) entities
-        pure (conns, cts, ucs, [], ms, sfts, rfts, pcs)
-      else do
-        withStore' unsetConnectionToSubscribe
-        (ctConns, cts) <- getContactConns
-        (ucConns, ucs) <- getUserContactLinkConns
-        (gs, mConns, ms) <- getGroupMemberConns
-        (sftConns, sfts) <- getSndFileTransferConns
-        (rftConns, rfts) <- getRcvFileTransferConns
-        (pcConns, pcs) <- getPendingContactConns
-        let conns = concat [ctConns, ucConns, mConns, sftConns, rftConns, pcConns]
-        pure (conns, cts, ucs, gs, ms, sfts, rfts, pcs)
-  -- subscribe using batched commands
-  rs <- withAgent $ \a -> agentBatchSubscribe a conns
-  -- send connection events to view
-  contactSubsToView rs cts ce
-  -- TODO possibly, we could either disable these events or replace with less noisy for API
-  contactLinkSubsToView rs ucs
-  groupSubsToView rs gs ms ce
-  sndFileSubsToView rs sfts
-  rcvFileSubsToView rs rfts
-  pendingConnSubsToView rs pcs
+  env <- ask
+  (conns, ctConns, ucs, gs, mConns, sfts, rfts, pcConns) <-
+    withStore' $ \db -> flip runReaderT env $ do
+      -- run in one transaction so onlyNeeded will be atomic
```

Comment:
it is unnecessary complication, as once the app goes into foreground NSE will no longer be executed, so onlyNeeded cannot change.

Even if it were not the case, making it atomic would not solve the problem, as if NSE could run _after_ the app re-opened, we would have a bigger issue - the new connections would not resubscribe.

So combining them into one transaction is completely unnecessary.